Add custom Groovy lexer to sphinx/rtd

elevans opened the pull request:
This PR adds a `lexers.py` module that contains a custom Groovy lexer that adds "#@" lines as comments. To use this custom lexer simply define your code block like so:

```text
.. code-block:: scijava-groovy
```

This PR resolves the annoying errors with "#@" in groovy scripts and stops the error highlighting in our scijava ops docs.

elevans committed:
Add custom Groovy lexer to sphinx/rtd

This commit adds a "lexer.py" module that contains
a custom Groovy lexer that adds "#@" lines as comments.
To use this custom lexer simply define your code block like so:

.. code-block:: fijigroovy

ctrueden commented on docs/conf.py:
```diff
@@ -56,6 +59,13 @@
 
+# -- Custom Lexers -----------------------------------------------------------
+
+from lexers import FijiGroovyLexer
```
Could we please call it `SciJavaGroovyLexer` instead? It really has nothing to do with Fiji specifically. The `#@` syntax is part of the SciJava scripting subsystem.

elevans replied:
Done!

ctrueden commented on docs/conf.py:
```diff
@@ -56,6 +59,13 @@
+from lexers import FijiGroovyLexer
+
+def setup(app):
+    app.add_lexer('fijigroovy', FijiGroovyLexer)
```
And similarly, here I would name the lexer `scijava-groovy`. With dash for clarity, assuming dashes are allowed. Or else `scijava_groovy` or `sciJavaGroovy` if dashes cannot be used.

elevans replied:
We can use dashes. The SciJava Groovy lexer alias is now `scijava-groovy`.

ctrueden reviewed (changes requested):
Thanks a lot for figuring this out! I would just request the name be "SciJava" instead of "Fiji" since that's the relevant layer here.

elevans committed:
Subclass the standard Groovy lexer

Instead of duplicating the Groovy lexer to
modify it, we can simply subclass it
and inherit it's tokens.

This commit also renames the lexer to
"SciJavaGroovyLexer".

elevans committed:
Rename fijigroovy lexer to scijava-groovy

elevans reviewed (commented):
Thanks for the feedback @ctrueden! We can use this custom Groovy lexer with the `scijava-groovy` alias.

ctrueden reviewed (approved):
Thanks @elevans! LGTM
